### PR TITLE
Remove duplicates in DNS cache list (By pre-traverse the cache table).

### PR DESCRIPTION
**Pull request by univert**
Also include the following updates:
Only traverse a small subset of DNS queries to boost performance.
Add a context menu to copy host name.

### CODE REVIEW

**Comment by univert** [comment]
My next step is to add an extra column in Network view list to show the original DNS query for a particular connection's remote address by consulting the dnscache table. This will be more informative than "Remote Hostname" which is usually some CDN server name.

File: DnsCachePlugin/main.c [hunk 1]
```diff
@@ -230,6 +275,15 @@ VOID ShowStatusMenu(
...
                 break;
+            case 2:
+                HGLOBAL hMem = GlobalAlloc(GMEM_MOVEABLE, cacheEntryName->Length * 2 + 2);
+                memcpy(GlobalLock(hMem), cacheEntryName->Buffer, cacheEntryName->Length*2);
+                GlobalUnlock(hMem);
+                OpenClipboard(0);
+                EmptyClipboard();
+                SetClipboardData(CF_UNICODETEXT, hMem);
+                CloseClipboard();
+                break;
```
Comment (dmex) [inline DnsCachePlugin/main.c]: PH has a helper function called PhSetClipboardString and you can replace this entire block of code with just this:
PhSetClipboardString(hwndDlg, &cacheEntryName->sr);

**Review by dmex — CHANGES_REQUESTED** [review]
Just needs the clipboard code replaced with PhSetClipboardString.

File: DnsCachePlugin/main.c (hunk 1 above)
Comment (univert) [reply DnsCachePlugin/main.c]: Done.

**Review by dmex — APPROVED** [review]

**Pull request merged**